incubator/buzzfeed-sso/templates/auth-deployment.yaml

```diff
@@ -0,0 +1,144 @@
+{{- $fullName := include "buzzfeed-sso.fullname" . -}}
+{{- $googleSecret := .Values.provider.google.customSecret | default (printf "%s-google" ($fullName)) -}}
+{{- $authSecret := .Values.auth.customSecret | default ($fullName) -}}
+{{- $name := include "buzzfeed-sso.name" . -}}
+{{- $authDomain := .Values.auth.domain -}}
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: {{ $fullName }}-auth
+  labels:
+    app: {{ $name }}
+    chart: {{ template "buzzfeed-sso.chart" . }}
+    component: {{ $name }}-auth
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.auth.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ $name }}
+      component: {{ $name }}-auth
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```

Comment:
@darioblanco the checksum should also contain something about the secrets so that when they change, the pods are restarted.

Reply:
I am thinking in reworking the checksum behavior and it would be useful to allow the user to specify annotations (for instance, I use fluentbit and would like to skip some logs).

Will work on it on this branch

Reply:
Fixed. In addition, now it is possible to configure annotations. This is quite handy for me, as I have already pushed my `fluentbit.io/exclude: "true"` annotation and I have checked that it works in my production cluster